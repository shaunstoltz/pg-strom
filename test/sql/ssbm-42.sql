SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q4_2
SET pg_strom.enabled = on;
select d_year, s_nation, p_category,
sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
  where lo_custkey = c_custkey
  and lo_suppkey = s_suppkey
  and lo_partkey = p_partkey
  and lo_orderdate = d_datekey
  and c_region = 'AMERICA'
  and s_region = 'AMERICA'
  and (d_year = 1997 or d_year = 1998)
  and (p_mfgr = 'MFGR#1'
   or p_mfgr = 'MFGR#2')
group by d_year, s_nation, p_category
order by d_year, s_nation, p_category;

SET pg_strom.enabled = off;
select d_year, s_nation, p_category,
sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
  where lo_custkey = c_custkey
  and lo_suppkey = s_suppkey
  and lo_partkey = p_partkey
  and lo_orderdate = d_datekey
  and c_region = 'AMERICA'
  and s_region = 'AMERICA'
  and (d_year = 1997 or d_year = 1998)
  and (p_mfgr = 'MFGR#1'
   or p_mfgr = 'MFGR#2')
group by d_year, s_nation, p_category
order by d_year, s_nation, p_category;
