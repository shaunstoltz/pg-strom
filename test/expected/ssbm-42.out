SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q4_2
SET pg_strom.enabled = on;
select d_year, s_nation, p_category,
sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
  where lo_custkey = c_custkey
  and lo_suppkey = s_suppkey
  and lo_partkey = p_partkey
  and lo_orderdate = d_datekey
  and c_region = 'AMERICA'
  and s_region = 'AMERICA'
  and (d_year = 1997 or d_year = 1998)
  and (p_mfgr = 'MFGR#1'
   or p_mfgr = 'MFGR#2')
group by d_year, s_nation, p_category
order by d_year, s_nation, p_category;
 d_year |    s_nation     | p_category |   profit   
--------+-----------------+------------+------------
   1997 | ARGENTINA       | MFGR#11    | 2205161194
   1997 | ARGENTINA       | MFGR#12    | 2157168886
   1997 | ARGENTINA       | MFGR#13    | 2199440157
   1997 | ARGENTINA       | MFGR#14    | 2212118831
   1997 | ARGENTINA       | MFGR#15    | 2199506700
   1997 | ARGENTINA       | MFGR#21    | 2163518018
   1997 | ARGENTINA       | MFGR#22    | 2227902674
   1997 | ARGENTINA       | MFGR#23    | 2259534188
   1997 | ARGENTINA       | MFGR#24    | 2220627198
   1997 | ARGENTINA       | MFGR#25    | 2148693785
   1997 | BRAZIL          | MFGR#11    | 2137375154
   1997 | BRAZIL          | MFGR#12    | 2148401817
   1997 | BRAZIL          | MFGR#13    | 2035891250
   1997 | BRAZIL          | MFGR#14    | 2226268805
   1997 | BRAZIL          | MFGR#15    | 1957374419
   1997 | BRAZIL          | MFGR#21    | 2001335694
   1997 | BRAZIL          | MFGR#22    | 2000408695
   1997 | BRAZIL          | MFGR#23    | 2206646971
   1997 | BRAZIL          | MFGR#24    | 2090537237
   1997 | BRAZIL          | MFGR#25    | 1942087586
   1997 | CANADA          | MFGR#11    | 2182320886
   1997 | CANADA          | MFGR#12    | 2141917083
   1997 | CANADA          | MFGR#13    | 2192696867
   1997 | CANADA          | MFGR#14    | 2114175777
   1997 | CANADA          | MFGR#15    | 2220385620
   1997 | CANADA          | MFGR#21    | 2113143726
   1997 | CANADA          | MFGR#22    | 2034476862
   1997 | CANADA          | MFGR#23    | 2277417590
   1997 | CANADA          | MFGR#24    | 2130021587
   1997 | CANADA          | MFGR#25    | 2055613005
   1997 | PERU            | MFGR#11    | 2271967110
   1997 | PERU            | MFGR#12    | 2269174347
   1997 | PERU            | MFGR#13    | 2184304126
   1997 | PERU            | MFGR#14    | 2337588948
   1997 | PERU            | MFGR#15    | 2163892151
   1997 | PERU            | MFGR#21    | 2249944016
   1997 | PERU            | MFGR#22    | 2203778623
   1997 | PERU            | MFGR#23    | 2139176272
   1997 | PERU            | MFGR#24    | 2390454604
   1997 | PERU            | MFGR#25    | 2025828600
   1997 | UNITED STATES   | MFGR#11    | 2211439868
   1997 | UNITED STATES   | MFGR#12    | 2000761099
   1997 | UNITED STATES   | MFGR#13    | 2119225152
   1997 | UNITED STATES   | MFGR#14    | 2040134118
   1997 | UNITED STATES   | MFGR#15    | 2282981185
   1997 | UNITED STATES   | MFGR#21    | 2191102462
   1997 | UNITED STATES   | MFGR#22    | 1909206776
   1997 | UNITED STATES   | MFGR#23    | 2174235942
   1997 | UNITED STATES   | MFGR#24    | 2054923784
   1997 | UNITED STATES   | MFGR#25    | 1964875246
   1998 | ARGENTINA       | MFGR#11    | 1251401348
   1998 | ARGENTINA       | MFGR#12    | 1222615645
   1998 | ARGENTINA       | MFGR#13    | 1219172107
   1998 | ARGENTINA       | MFGR#14    | 1348364764
   1998 | ARGENTINA       | MFGR#15    | 1312181321
   1998 | ARGENTINA       | MFGR#21    | 1207998927
   1998 | ARGENTINA       | MFGR#22    | 1344643128
   1998 | ARGENTINA       | MFGR#23    | 1343299602
   1998 | ARGENTINA       | MFGR#24    | 1285099752
   1998 | ARGENTINA       | MFGR#25    | 1310478526
   1998 | BRAZIL          | MFGR#11    | 1152607789
   1998 | BRAZIL          | MFGR#12    | 1085177272
   1998 | BRAZIL          | MFGR#13    | 1125914127
   1998 | BRAZIL          | MFGR#14    | 1163292828
   1998 | BRAZIL          | MFGR#15    | 1156050633
   1998 | BRAZIL          | MFGR#21    | 1121524292
   1998 | BRAZIL          | MFGR#22    | 1266409130
   1998 | BRAZIL          | MFGR#23    | 1178677345
   1998 | BRAZIL          | MFGR#24    | 1303324259
   1998 | BRAZIL          | MFGR#25    | 1157975835
   1998 | CANADA          | MFGR#11    | 1377322070
   1998 | CANADA          | MFGR#12    | 1190972181
   1998 | CANADA          | MFGR#13    | 1247292823
   1998 | CANADA          | MFGR#14    | 1125287847
   1998 | CANADA          | MFGR#15    | 1309302909
   1998 | CANADA          | MFGR#21    | 1232164990
   1998 | CANADA          | MFGR#22    | 1264367329
   1998 | CANADA          | MFGR#23    | 1209249829
   1998 | CANADA          | MFGR#24    | 1166375506
   1998 | CANADA          | MFGR#25    | 1183505275
   1998 | PERU            | MFGR#11    | 1108095403
   1998 | PERU            | MFGR#12    | 1180758517
   1998 | PERU            | MFGR#13    | 1293619520
   1998 | PERU            | MFGR#14    | 1456648538
   1998 | PERU            | MFGR#15    | 1146516776
   1998 | PERU            | MFGR#21    | 1155446821
   1998 | PERU            | MFGR#22    | 1268011297
   1998 | PERU            | MFGR#23    | 1307789985
   1998 | PERU            | MFGR#24    | 1187438414
   1998 | PERU            | MFGR#25    | 1434753194
   1998 | UNITED STATES   | MFGR#11    | 1230816937
   1998 | UNITED STATES   | MFGR#12    | 1193303320
   1998 | UNITED STATES   | MFGR#13    | 1167915641
   1998 | UNITED STATES   | MFGR#14    | 1253459724
   1998 | UNITED STATES   | MFGR#15    | 1296490693
   1998 | UNITED STATES   | MFGR#21    | 1137340543
   1998 | UNITED STATES   | MFGR#22    | 1206598406
   1998 | UNITED STATES   | MFGR#23    | 1251349168
   1998 | UNITED STATES   | MFGR#24    | 1172918416
   1998 | UNITED STATES   | MFGR#25    | 1203882368
(100 rows)

SET pg_strom.enabled = off;
select d_year, s_nation, p_category,
sum(lo_revenue - lo_supplycost) as profit
from date1, customer, supplier, part, lineorder
  where lo_custkey = c_custkey
  and lo_suppkey = s_suppkey
  and lo_partkey = p_partkey
  and lo_orderdate = d_datekey
  and c_region = 'AMERICA'
  and s_region = 'AMERICA'
  and (d_year = 1997 or d_year = 1998)
  and (p_mfgr = 'MFGR#1'
   or p_mfgr = 'MFGR#2')
group by d_year, s_nation, p_category
order by d_year, s_nation, p_category;
 d_year |    s_nation     | p_category |   profit   
--------+-----------------+------------+------------
   1997 | ARGENTINA       | MFGR#11    | 2205161194
   1997 | ARGENTINA       | MFGR#12    | 2157168886
   1997 | ARGENTINA       | MFGR#13    | 2199440157
   1997 | ARGENTINA       | MFGR#14    | 2212118831
   1997 | ARGENTINA       | MFGR#15    | 2199506700
   1997 | ARGENTINA       | MFGR#21    | 2163518018
   1997 | ARGENTINA       | MFGR#22    | 2227902674
   1997 | ARGENTINA       | MFGR#23    | 2259534188
   1997 | ARGENTINA       | MFGR#24    | 2220627198
   1997 | ARGENTINA       | MFGR#25    | 2148693785
   1997 | BRAZIL          | MFGR#11    | 2137375154
   1997 | BRAZIL          | MFGR#12    | 2148401817
   1997 | BRAZIL          | MFGR#13    | 2035891250
   1997 | BRAZIL          | MFGR#14    | 2226268805
   1997 | BRAZIL          | MFGR#15    | 1957374419
   1997 | BRAZIL          | MFGR#21    | 2001335694
   1997 | BRAZIL          | MFGR#22    | 2000408695
   1997 | BRAZIL          | MFGR#23    | 2206646971
   1997 | BRAZIL          | MFGR#24    | 2090537237
   1997 | BRAZIL          | MFGR#25    | 1942087586
   1997 | CANADA          | MFGR#11    | 2182320886
   1997 | CANADA          | MFGR#12    | 2141917083
   1997 | CANADA          | MFGR#13    | 2192696867
   1997 | CANADA          | MFGR#14    | 2114175777
   1997 | CANADA          | MFGR#15    | 2220385620
   1997 | CANADA          | MFGR#21    | 2113143726
   1997 | CANADA          | MFGR#22    | 2034476862
   1997 | CANADA          | MFGR#23    | 2277417590
   1997 | CANADA          | MFGR#24    | 2130021587
   1997 | CANADA          | MFGR#25    | 2055613005
   1997 | PERU            | MFGR#11    | 2271967110
   1997 | PERU            | MFGR#12    | 2269174347
   1997 | PERU            | MFGR#13    | 2184304126
   1997 | PERU            | MFGR#14    | 2337588948
   1997 | PERU            | MFGR#15    | 2163892151
   1997 | PERU            | MFGR#21    | 2249944016
   1997 | PERU            | MFGR#22    | 2203778623
   1997 | PERU            | MFGR#23    | 2139176272
   1997 | PERU            | MFGR#24    | 2390454604
   1997 | PERU            | MFGR#25    | 2025828600
   1997 | UNITED STATES   | MFGR#11    | 2211439868
   1997 | UNITED STATES   | MFGR#12    | 2000761099
   1997 | UNITED STATES   | MFGR#13    | 2119225152
   1997 | UNITED STATES   | MFGR#14    | 2040134118
   1997 | UNITED STATES   | MFGR#15    | 2282981185
   1997 | UNITED STATES   | MFGR#21    | 2191102462
   1997 | UNITED STATES   | MFGR#22    | 1909206776
   1997 | UNITED STATES   | MFGR#23    | 2174235942
   1997 | UNITED STATES   | MFGR#24    | 2054923784
   1997 | UNITED STATES   | MFGR#25    | 1964875246
   1998 | ARGENTINA       | MFGR#11    | 1251401348
   1998 | ARGENTINA       | MFGR#12    | 1222615645
   1998 | ARGENTINA       | MFGR#13    | 1219172107
   1998 | ARGENTINA       | MFGR#14    | 1348364764
   1998 | ARGENTINA       | MFGR#15    | 1312181321
   1998 | ARGENTINA       | MFGR#21    | 1207998927
   1998 | ARGENTINA       | MFGR#22    | 1344643128
   1998 | ARGENTINA       | MFGR#23    | 1343299602
   1998 | ARGENTINA       | MFGR#24    | 1285099752
   1998 | ARGENTINA       | MFGR#25    | 1310478526
   1998 | BRAZIL          | MFGR#11    | 1152607789
   1998 | BRAZIL          | MFGR#12    | 1085177272
   1998 | BRAZIL          | MFGR#13    | 1125914127
   1998 | BRAZIL          | MFGR#14    | 1163292828
   1998 | BRAZIL          | MFGR#15    | 1156050633
   1998 | BRAZIL          | MFGR#21    | 1121524292
   1998 | BRAZIL          | MFGR#22    | 1266409130
   1998 | BRAZIL          | MFGR#23    | 1178677345
   1998 | BRAZIL          | MFGR#24    | 1303324259
   1998 | BRAZIL          | MFGR#25    | 1157975835
   1998 | CANADA          | MFGR#11    | 1377322070
   1998 | CANADA          | MFGR#12    | 1190972181
   1998 | CANADA          | MFGR#13    | 1247292823
   1998 | CANADA          | MFGR#14    | 1125287847
   1998 | CANADA          | MFGR#15    | 1309302909
   1998 | CANADA          | MFGR#21    | 1232164990
   1998 | CANADA          | MFGR#22    | 1264367329
   1998 | CANADA          | MFGR#23    | 1209249829
   1998 | CANADA          | MFGR#24    | 1166375506
   1998 | CANADA          | MFGR#25    | 1183505275
   1998 | PERU            | MFGR#11    | 1108095403
   1998 | PERU            | MFGR#12    | 1180758517
   1998 | PERU            | MFGR#13    | 1293619520
   1998 | PERU            | MFGR#14    | 1456648538
   1998 | PERU            | MFGR#15    | 1146516776
   1998 | PERU            | MFGR#21    | 1155446821
   1998 | PERU            | MFGR#22    | 1268011297
   1998 | PERU            | MFGR#23    | 1307789985
   1998 | PERU            | MFGR#24    | 1187438414
   1998 | PERU            | MFGR#25    | 1434753194
   1998 | UNITED STATES   | MFGR#11    | 1230816937
   1998 | UNITED STATES   | MFGR#12    | 1193303320
   1998 | UNITED STATES   | MFGR#13    | 1167915641
   1998 | UNITED STATES   | MFGR#14    | 1253459724
   1998 | UNITED STATES   | MFGR#15    | 1296490693
   1998 | UNITED STATES   | MFGR#21    | 1137340543
   1998 | UNITED STATES   | MFGR#22    | 1206598406
   1998 | UNITED STATES   | MFGR#23    | 1251349168
   1998 | UNITED STATES   | MFGR#24    | 1172918416
   1998 | UNITED STATES   | MFGR#25    | 1203882368
(100 rows)

